### PR TITLE
Move bypass execution policy check after AppLocker Deny check

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Move bypass execution policy check after AppLocker Deny check

## PR Context

This only affects a corner case where you only have Deny rules.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/security/SecurityManager.cs [hunk 1]
```diff
@@ -184,6 +180,11 @@ private bool CheckPolicy(ExternalScriptInfo script, PSHost host, out Exception r
                 }
             }
 
+            // WLDP and Applocker takes priority over powershell exeuction policy.
+            // See if they want to bypass the authorization manager
+            if (executionPolicy == ExecutionPolicy.Bypass)
```
Comment (PaulHigin) [inline src/System.Management.Automation/security/SecurityManager.cs]: Nitpick.  I believe our style guidelines say the conditional statement must be within brackets.
```suggestion
            if (executionPolicy == ExecutionPolicy.Bypass)
            {
                return true;
            }
```

**Review by PaulHigin — APPROVED** [review]
Minor comment.

**Review by iSazonov — APPROVED** [review]

**Review by rjmholt — APPROVED** [review]

**Comment by rjmholt** [comment]
@PaulHigin can you refresh your review?

**Review by PaulHigin — APPROVED** [review]

**Comment by strawgate** [comment]
This seems like it would be a breaking change....

**Comment by iSazonov** [comment]
> This seems like it would be a breaking change

It is _a bug fix_ - no one can disable the lock set by a system administrator.

**Comment by strawgate** [comment]
Does this being a bugfix effect whether it's a breaking change?

https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/breaking-change-contract.md

> It doesn't matter if the existing behavior is "wrong", we still need to think through the implications. PowerShell has been in broad use for over 10 years so we be must be extremely sensitive to breaking existing users and scripts.

**Comment by iSazonov** [comment]
@strawgate Any bug fix could be considered as a breaking change :-)

This is a security related bug fix - such things are never discussed publicly.

**Comment by TravisEz13** [comment]
@strawgate I've marked the issue a breaking change.  As @iSazonov has said, AppLocker is considered the security feature with the stronger promise here.  Execution policy was never intended as a bypass.

See our security policy:  https://github.com/PowerShell/PowerShell/security/policy
> Security issues are treated very seriously and will, by default, takes precedence over other considerations including usability, performance, etc... Best effort will be used to mitigate side effects of a security change, but PowerShell must be secure by default.

**Pull request merged**